test/src/test/java/hudson/diagnosis/OldDataMonitorTest.java

```diff
@@ -149,6 +156,21 @@ public Void call() throws Exception {
 
     }
 
+    @Issue("JENKINS-26718")
+    @Test public void unlocatableRun() throws Exception {
+        OldDataMonitor odm = OldDataMonitor.get(r.jenkins);
+        FreeStyleProject p = mock(FreeStyleProject.class);
+        when(p.getParent()).thenReturn(Jenkins.getInstance());
+        when(p.getFullName()).thenReturn("notfound");
+        FreeStyleBuild build = new FreeStyleBuild(p);
+        odm.report(build, (String) null);
+
+        assertEquals(Collections.singleton(build), odm.getData().keySet());
+        odm.doDiscard(null, null);
+        assertEquals(0, odm.getData().keySet().size());
```

Comment:
Alternately, a bit nicer in case it ever fails:

    assertEquals(Collections.emptySet(), odm.getData().keySet());